### PR TITLE
Sync with nix.dev

### PR DESCRIPTION
**Pull request by garbas**
- Add missing tutorials from nix.dev
- Move `Install Nix` tutorial to be included in Download page
- Fix link to PatchELF

### CODE REVIEW

File: download.tt [hunk 1]
```diff
@@ -16,23 +16,104 @@
...
+          Install Nix on via the
+          <strong>recommended</strong>
+          <a href="[%nixManual%]/installation/multi-user.html">multi-user installation</a>:
         </p>
         <pre class="terminal-console">
-<span class="shell-prompt">$ </span>curl -L https://nixos.org/nix/install | sh</pre>
-        <p>Make sure to follow the instructions output by the script.</p>
-        <p>The installation script requires that you have <code>sudo</code> access to <code>root</code>.</p>
+<span class="shell-prompt">$ </span>sh &lt;(curl -L https://nixos.org/nix/install) \
+        --darwin-use-unencrypted-nix-store-volume \
```
Comment (garbas) [inline download.tt:72]: - Explain the options
- Mentions why no single user

**Comment by garbas** [comment]
Add "More ..." section to NixOS portion of the download page

File: download.tt [hunk 2]
```diff
@@ -16,23 +16,104 @@
...
+        <p>
+          Install Nix on via the
+          <a href="[%nixManual%]/installation/single-user.html">single-user installation</a>:
+        </p>
+        <pre class="terminal-console">
+<span class="shell-prompt">$ </span>sh &lt;(curl -L https://nixos.org/nix/install) --no-daemon</pre>
+
+        <h3>Which type of installation should you choose?</h3>
+        <p>
+          This boild down to your requirements, but here is a short list of
```
Comment (tomberek) [inline download.tt:44]:
```suggestion
          This boiled down to your requirements, but here is a short list of
```

File: download.tt [hunk 3]
```diff
@@ -16,23 +16,104 @@
...
   <div class="collapse">
     <div>
-      <article id="nix-quick-install">
-        <h2>Quick install</h2>
+      <article id="nix-install-linux">
+        <h2>Linux</h2>
+
+        <h3>Multi-user installation (recommended)</h3>
+        <p>
+          Install Nix on via the
```
Comment (tomberek) [inline download.tt:27]:
```suggestion
          Install Nix via the
```

File: download.tt [hunk 4]
```diff
@@ -16,23 +16,104 @@
...
+          <strong>recommended</strong>
+          <a href="[%nixManual%]/installation/multi-user.html">multi-user installation</a>:
+        </p>
+        <pre class="terminal-console">
+<span class="shell-prompt">$ </span>sh &lt;(curl -L https://nixos.org/nix/install) --daemon</pre>
+
+        <h3>Single-user installation</h3>
+        <p>
+          Install Nix on via the
```
Comment (tomberek) [inline download.tt:36]:
```suggestion
          Install Nix via the
```

File: download.tt [hunk 5]
```diff
@@ -16,23 +16,104 @@
...
+        <h3>Which type of installation should you choose?</h3>
+        <p>
+          This boild down to your requirements, but here is a short list of
+          reasons why <strong>we recommend</strong>
+          <a href="[%nixManual%]/installation/multi-user.html">multi-user installation</a>:
+        </p>
+
+        <h5>Pros</h5>
+        <ul>
+          <li>Better build isolation (and that is what Nix all about)</li>
```
Comment (tomberek) [inline download.tt:51]:
```suggestion
          <li>Better build isolation (and that is what Nix is all about)</li>
```

File: download.tt [hunk 6]
```diff
@@ -16,23 +16,104 @@
...
+          <li>Harder to uninstall</li>
+        </ul>
+      </article>
+
+      <article id="nix-install-macosx">
+        <h2>MacOS</h2>
         <p>
-          The quickest way to install Nix is to open a terminal and run the following
-          command (as a user other than <code>root</code> with <code>sudo</code> permission):
+          Install Nix on via the
```
Comment (tomberek) [inline download.tt:66]:
```suggestion
          Install Nix via the
```

File: download.tt [hunk 7]
```diff
@@ -16,23 +16,104 @@
...
+<span cls="shell-prompt">$ </span>docker run -it -v $(pwd)/nixpkgs:/nixpkgs nixos/nix
+<span cls="shell-prompt">docker&gt; </span>nix-build -I nixpkgs=/nixpkgs -A hello
+<span cls="shell-prompt">docker&gt; </span>find ./result # this symlink points to the build package</pre>
+      </article>
+
       <article id="nix-verify-installation">
         <h2>Verify installation</h2>
-        <p>You may want to verify the integrity of the installation script using GPG:</p>
+        <p>
+          Check that the installation by opening
```
Comment (tomberek) [inline download.tt:106]:
```suggestion
          Check the installation by opening
```

**Pull request merged**